meta-mender-tegra/scripts/mender_tegra_test.py

```diff
@@ -0,0 +1,263 @@
+        args=self.get_args()
+        """
+        Returns True if host (str) responds to a ping request.
+        Remember that a host may not respond to a ping (ICMP) request even if the host name is valid.
+        See https://stackoverflow.com/a/32684938/1446624
+        """
+
+        # Option for the number of packets as a function of
+        param = '-n' if platform.system().lower() == 'windows' else '-c'
+
+        # Building the command. Ex: "ping -c 1 google.com"
+        command = ['ping', param, '1', args.device]
+
+        return subprocess.call(command, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT) == 0
+
+    def wait_for_device_removal(self):
+        while self.ping():
+            pass
+
+    def mender_install(self):
+        args = self.get_args()
+        conn = self.get_connection()
+        if args.install is None:
+            self.get_parser().print_help()
+            print("Missing argument install")
+            raise RuntimeError("Missing argument install")
+        result = conn.run("mender -install {}".format(args.install))
+        # Mender doesn't return error states, so we can't check return codes here
+        match = re.search(r' level=error ', result.stderr, re.MULTILINE)
+        if match is not None:
+            raise RuntimeError("Mender install failed with error messages in the logs")
+
+    def add_sentinel_file(self):
+        conn = self.get_connection()
+        conn.run("mkdir -p /var/lib/mender")
+        conn.run("touch /var/lib/mender/dont-mark-next-boot-successful")
+
+    def remove_sentinel_file(self):
+        conn = self.get_connection()
+        conn.run("rm /var/lib/mender/dont-mark-next-boot-successful")
+
+    def mender_commit(self):
+        self.get_connection().run("mender -commit")
+
+    def reboot(self):
+        conn = self.get_connection()
+        print("Rebooting device")
+        result = conn.run("reboot", warn=True)
+        self.wait_for_device_removal()
+        self.wait_for_device()
+
+    def nvbootctrl_current_slot(self):
+        conn = self.get_connection()
+        result = conn.run("nvbootctrl get-current-slot")
+        return result.stdout.strip()
+
+    def check_partition_mismatch(self):
+        conn = self.get_connection()
+        boot_slot = self.nvbootctrl_current_slot()
+        rootfs_part = None
+        if boot_slot == '0':
+            rootfs_part = 'A'
+        elif boot_slot == '1':
+            rootfs_part = 'B'
+
+        if rootfs_part != None:
+            result = conn.run(f'grep -h {rootfs_part} /etc/mender/mender.conf /var/lib/mender/mender.conf | cut -d: -f2 | cut -d, -f1 | tr -d \'" \'')
+            result = conn.run(f'df -h | grep {result.stdout.strip()}')
+            if result.return_code != 0:
+                raise RuntimeError("Boot and Rootfs Partition Mismatch detected")
+        else:
+            raise RuntimeError("Cannot Identify Rootfs Partition Slot")
+
+    def verify_slot_change_across_reboots(self):
```

Comment:
@jajoosiddhant suggest "reboot_and_verify_slot" for the name here.
Also I think you can consolidate this function with the next function since the only difference is the error message and you can tell where it failed from the stacktrace.

Reply:
@dwalkes The main difference is in the if condition to check if the boot slots are equal or not. Having separate functions may give us flexibility for future use cases if required.
`reboot_and_verify_slot` might be confusing since we aren't rebooting in that function but checking slot change after a reboot using the function?

Reply:
actually nevermind the name looks good